### PR TITLE
Re-enable SET AUTHORIZATION ROLE on VIEW SECURITY DEFINER

### PR DESCRIPTION
**Pull request by ksobolew**
This PR re-enables `SET AUTHORIZATION ROLE` on views with `SECURITY DEFINER` with new access control rules that are meant to prevent any potential mis-use.

The new rules are:

* `SECURITY INVOKER`: unchanged; the view owner can change ownership to any user or role
* `SECURITY DEFINER`: as above, plus additional checks that depend on whether the target principal is USER or ROLE:
  * `USER`: disallowed (ideally there should be a separate privilege for this)
  * `ROLE`: the change is allowed if both of the following checks pass:
    1. The current user would be able to create the view from scratch (has privileges to create views in the catalog, is able to `SELECT` all tables and `EXECUTE` all functions accessed by the view)
    1. The target principal has privileges to create views in the catalog

Basically the idea is that it should be allowed if it would also be possible to just `DROP` and `CREATE` the view from scratch with the new owner - but only with the view is executed with the definer's privileges, because ownership does not affect what the view can access when it's executed with privileges of the invoker.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/execution/SetViewAuthorizationTask.java [hunk 1]
```diff
@@ -80,16 +110,115 @@ public String getName()
...
+    private void checkCanAccessViewEntities(Session session, QueryStateMachine stateMachine, List<Expression> parameters, QualifiedObjectName viewName, ViewDefinition view, TrinoPrincipal targetPrincipal)
+    {
+        CreateView statement = new CreateView(
+                QualifiedName.of(viewName.getCatalogName(), viewName.getSchemaName(), viewName.getObjectName()),
+                (Query) sqlParser.createStatement(view.getOriginalSql(), createParsingOptions(session)),
+                false,
+                view.getComment(),
+                Optional.of(DEFINER));
+        try {
+            analyzerFactory.createAnalyzer(session, parameters, parameterExtractor(statement, parameters), stateMachine.getWarningCollector())
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/execution/SetViewAuthorizationTask.java:172]: don't you need to use `targetPrincipal` as session identity? Because you verify if current identity can create a view and access view entities, not the one who will be the "owner"

File: core/trino-main/src/main/java/io/trino/execution/SetViewAuthorizationTask.java [hunk 2]
```diff
@@ -80,16 +110,115 @@ public String getName()
...
+                    .analyze(statement);
+        }
+        catch (AccessDeniedException e) {
+            wrapAccessDeniedException(viewName, targetPrincipal, e);
+        }
+    }
+
+    private void checkTargetPrincipalCanCreateView(Session session, TransactionManager transactionManager, AccessControl accessControl, QualifiedObjectName viewName, TrinoPrincipal targetPrincipal)
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/execution/SetViewAuthorizationTask.java:180]: Do we need this if we have `checkCanAccessViewEntities` which is using `CreateView` statement so I guess `checkCanCreateView` is called already there.

**Comment by mosabua** [comment]
👋 @ksobolew is this still in progress or should we close the PR.

**Comment by ksobolew** [comment]
> 👋 @ksobolew is this still in progress or should we close the PR.

No, it looks like we will keep this disabled for the foreseeable future. I haven't heard anyone complaining about this, so...
